registration/test/registration/services/WindowsNotificationRegistrarSpec.scala

```diff
@@ -19,75 +19,94 @@ with Mockito {
-      val channelUri = hubRegResponse.channelUri
-      val lastKnownChannelUri = "lastKnownChannelUri"
-      hubClient.registrationsByChannelUri(lastKnownChannelUri) returns Future.successful(List(hubRegResponse.copy(channelUri = lastKnownChannelUri)).right)
-      hubClient.registrationsByChannelUri(channelUri) returns Future.successful(Nil.right)
+    "update existing registration when registration with same userId already exist" in new registrations {
+      val channelUri = registration.deviceId
+      hubClient.registrationsByTag(userIdTag) returns Future.successful(List(hubRegResponse).right)
       hubClient.update(hubRegResponse.registration, fromMobileRegistration(registration)) returns Future.successful(hubRegResponse.right)
 
-      val response = provider.register(lastKnownChannelUri, registration)
+      val response = provider.register(channelUri, registration)
 
-      response must beEqualTo(RegistrationResponse(
-        deviceId = channelUri,
-        platform = WindowsMobile,
-        userId = registration.userId,
-        topics = Set.empty
-      ).right).await
+      response must beEqualTo(registrationResponse.right).await
+      there was no(hubClient).create(any[RawWindowsRegistration])
+      there was one(hubClient).update(any[WNSRegistrationId], any[RawWindowsRegistration])
+      there was no(hubClient).delete(any[WNSRegistrationId])
     }
 
-    "update existing registration, if the target registration already exists (instead of creating a new one)" in new registrations {
+    "update existing registration, including channelUri when the registration already exist" in new registrations {
       val channelUri = hubRegResponse.channelUri
       val lastKnownChannelUri = "lastKnownChannelUri"
       hubClient.registrationsByChannelUri(lastKnownChannelUri) returns Future.successful(List(hubRegResponse.copy(channelUri = lastKnownChannelUri)).right)
-      hubClient.registrationsByChannelUri(channelUri) returns Future.successful(List(hubRegResponse).right)
+      hubClient.registrationsByTag(userIdTag) returns Future.successful(Nil.right)
       hubClient.update(hubRegResponse.registration, fromMobileRegistration(registration)) returns Future.successful(hubRegResponse.right)
 
       val response = provider.register(lastKnownChannelUri, registration)
 
-      response must beEqualTo(RegistrationResponse(
-        deviceId = channelUri,
-        platform = WindowsMobile,
-        userId = registration.userId,
-        topics = Set.empty
-      ).right).await
+      response must beEqualTo(registrationResponse.right).await
+      there was no(hubClient).create(any[RawWindowsRegistration])
+      there was one(hubClient).update(any[WNSRegistrationId], any[RawWindowsRegistration])
+      there was no(hubClient).delete(any[WNSRegistrationId])
+    }
+
+    "delete all and replace by only one registration if more than one notification for the same userId" in new registrations {
```

Comment:
shouldn't be `...than one registration for the...` instead `..than one notification for the..` ?